### PR TITLE
zboss: libraries update

### PR DESCRIPTION
**Pull request by edmont**
ZBOSS libraries updates:

- Remove ZBOSS development libraries and deprecate `ZIGBEE_LIBRARY_DEVELOPMENT` configuration.
- Update ZBOSS production libraries with bugfixes.

### CODE REVIEW

**Comment by edmont** [comment]
Opening this PR as draft with the intention of extending it with bugfixes update and only then trigger the testing.

@greg-fer or @wiba-nordic please take a look to the doc changes in case another approach would be preferred.

File: zboss/README.rst [hunk 1]
```diff
@@ -5,20 +5,14 @@ ZBOSS Zigbee stack
 
 The |NCS|'s :ref:`nrf:ug_zigbee` stack uses ZBOSS – a portable, high-performance Zigbee software protocol stack that allows for interoperability, customizing, testing, and optimizing of your Zigbee solution.
 
-The nrfxlib repository contains the following versions of the ZBOSS libraries:
+The nrfxlib repository contains the following version of the ZBOSS libraries:
 
 * *Production* version that contains the latest stable ZBOSS libraries.
```
Comment (wiba-nordic) [inline zboss/README.rst:10]: I don't recall there being that many ZBOSS updates lately, so is saying that they contain the latest stable libraries correct?

File: zboss/doc/zboss_configuration.rst [hunk 2]
```diff
@@ -11,7 +11,6 @@ The ZBOSS stack is distributed as a set of precompiled linkable libraries.
 These libraries are available in the following versions:
 
 * `ZBOSS production libraries`_
```
Comment (wiba-nordic) [inline zboss/doc/zboss_configuration.rst:13]:
```suggestion
The available libraries are the `ZBOSS production libraries`_.
```

File: zboss/doc/zboss_configuration.rst [hunk 3]
```diff
@@ -11,7 +11,6 @@ The ZBOSS stack is distributed as a set of precompiled linkable libraries.
...
-* `ZBOSS development libraries`_
 
 Each version supports different Zigbee device roles, with each variant having its own configuration.
 This allows you to scale the application and select the most suitable set of features.
```
Comment (wiba-nordic) [inline zboss/doc/zboss_configuration.rst:16]: I feel like these lines don't need to be here now that we only offer one set of libraries.

**Review by wiba-nordic — CHANGES_REQUESTED** [review]
I have no problem with this approach.

File: zboss/README.rst (hunk 1 above)
Comment (edmont) [reply zboss/README.rst:10]: The intention is to update the libraries in this same PR.

File: zboss/doc/zboss_configuration.rst (hunk 3 above)
Comment (edmont) [reply zboss/doc/zboss_configuration.rst:16]: But within the set of libraries there are still variants.

**Review by greg-fer — COMMENTED** [review]
Are you planning any updates to https://docs.nordicsemi.com/bundle/ncs-latest/page/nrfxlib/zboss/README.html and its documentation https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/index.html ?

**Comment by edmont** [comment]
> Are you planning any updates to https://docs.nordicsemi.com/bundle/ncs-latest/page/nrfxlib/zboss/README.html and its documentation https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/index.html ?

Yes, they should be updated.

File: zboss/doc/zboss_configuration.rst [hunk 4]
```diff
@@ -8,10 +8,7 @@ ZBOSS library configuration
...
 
 The ZBOSS stack is distributed as a set of precompiled linkable libraries.
-These libraries are available in the following versions:
-
-* `ZBOSS production libraries`_
-* `ZBOSS development libraries`_
+The available libraries are the `ZBOSS production libraries`_.
 
 Each version supports different Zigbee device roles, with each variant having its own configuration.
 This allows you to scale the application and select the most suitable set of features.
```
Comment (wiba-nordic) [inline zboss/doc/zboss_configuration.rst:14]:
```suggestion
The libraries supports different Zigbee device roles, with each variant having its own configuration.
This allows you to scale the application and select the most suitable set of features.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

**Comment by greg-fer** [comment]
> > Are you planning any updates to https://docs.nordicsemi.com/bundle/ncs-latest/page/nrfxlib/zboss/README.html and its documentation https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/index.html ?
> 
> Yes, they should be updated.

OK, can you point me to the PR in Bitbucket with the integration of the new version of the ZBOSS stack? Thanks.

File: zboss/CHANGELOG.rst [hunk 5]
```diff
@@ -9,6 +9,31 @@ Changelog
...
+
+* Precompiled development ZBOSS libraries.
+
+Bug fixes
+=========
+
+* Fixed security vulnerabilities.
+* Fixed an issue with group command with group ID 0xFFF7.
+* Fixed an issue with incorrect access type in Zigbee Discover Attributes Extended Response.
+* Fixed an issue with incorrect handling two "move to color" commands received in a row.
```
Comment (wiba-nordic) [inline zboss/CHANGELOG.rst:32]:
```suggestion
* Fixed an issue with incorrect handling of two "move to color" commands received in a row.
```

File: zboss/CHANGELOG.rst [hunk 6]
```diff
@@ -9,6 +9,31 @@ Changelog
...
+* Fixed an issue with incorrect validation of Manufacturer specific attribute bit.
```
Comment (wiba-nordic) [inline zboss/CHANGELOG.rst:33]:
```suggestion
* Fixed an issue with incorrect validation of Manufacturer-specific attribute bit.
```

File: zboss/CHANGELOG.rst [hunk 7]
```diff
@@ -9,6 +9,31 @@ Changelog
...
+* Fixed an issue with incorrect response of ZC to beacon when joining is open.
+* Fixed an issue that prevented to define individual reporting contexts when multiple endpoints are implemented.
```
Comment (wiba-nordic) [inline zboss/CHANGELOG.rst:35]:
```suggestion
* Fixed an issue that prevented defining individual reporting contexts when multiple endpoints are implemented.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

**Review by wiba-nordic — APPROVED** [review]
Documentation looks good to me.

**Comment by edmont** [comment]
Just removed previously added unneeded libraries.

**Comment by edmont** [comment]
Setting DNM while performing thorough testing.

**Comment by jfischer-no** [comment]
:hourglass_flowing_sand: @edmont Please ping me when it is ready.

**Review by AdamZelikNS — APPROVED** [review]

**Pull request merged**